plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReLoader.java

```diff
@@ -0,0 +1,285 @@
+    ) {
+        this.clusterService = clusterService;
+        this.client = client;
+        this.threadPool = threadPool;
+        this.xContentRegistry = xContentRegistry;
+        this.nodeHelper = nodeHelper;
+        this.mlStats = mlStats;
+
+        enableAutoReLoadModel = ML_COMMONS_MODEL_AUTO_RELOAD_ENABLE.get(settings);
+        clusterService
+            .getClusterSettings()
+            .addSettingsUpdateConsumer(ML_COMMONS_MODEL_AUTO_RELOAD_ENABLE, it -> enableAutoReLoadModel = it);
+    }
+
+    /**
+     * the main method: model auto reloading
+     */
+    public void autoReLoadModel() {
+        log.debug("enableAutoReLoadModel {} ", enableAutoReLoadModel);
+
+        // if we don't need to reload automatically, just return without doing anything
+        if (!enableAutoReLoadModel) {
+            return;
+        }
+
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            // At opensearch startup, get local node id, if not ml node,we ignored, just return without doing anything
+            String localNodeId = clusterService.localNode().getId();
+            DiscoveryNode node = nodeHelper.getNode(localNodeId);
+            if (!MLNodeUtils.isMLNode(node)) {
+                return;
+            }
+
+            // According to the node id to query the number of retries, if more than 2 (the maximum number of retries), we don't need to
+            // retry,
+            // that the number of unsuccessful reload has reached the maximum number of times, do not need to reload
+            Integer reTryTimes = 0;
+            if (isExistedIndex(ML_MODEL_RELOAD_INDEX)) {
+                reTryTimes = getReTryTimes(localNodeId);
+            }
+            if (reTryTimes > ML_MODEL_RELOAD_MAX_RETRY_TIMES) {
+                log.debug("have exceeded max retry times, always failure");
+            }
+
+            // auto reload all models of this local node, if it fails, reTryTimes+1, if it succeeds, reTryTimes is cleared to 0
+            try {
+                CompletableFuture<Void> completableFuture = CompletableFuture.runAsync(() -> autoReLoadModelByNodeId(localNodeId));
+                completableFuture.get();
+            } catch (InterruptedException | ExecutionException e) {
+                throw new RuntimeException("Can't auto reload model, the reason is:", e);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("Can't auto reload model");
+        }
+    }
+
+    /**
+     * auto reload all the models under the node id
+     * the node must be a ml node
+     * @param nodeId node id
+     */
+    @VisibleForTesting
+    void autoReLoadModelByNodeId(String nodeId) {
+        if (!isExistedIndex(ML_TASK_INDEX)) {
+            return;
+        }
+
+        SearchRequest searchRequest = new SearchRequest(ML_TASK_INDEX);
+        SearchResponse response = client.execute(SearchAction.INSTANCE, searchRequest).actionGet();
+
```

Comment:
Three checks can be merged into one if check.

Reply:
` if (!isExistedIndex(ML_TASK_INDEX)) {
            return;
        }`
this check shouldn't be removed, because I found the code will throw the  IndexNotFoundException if I didn't check whether ML_TASK_INDEX exised.
And other checks have been merged into 1 by me